contracts/Person.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.4.24;
+
+import "./interfaces/ERC20Interface.sol";
+import "../lib/ds-proxy/src/proxy.sol";
+
+contract Person is DSProxy {
+  // TODO: Add limits?
+  mapping (address => bool) public isEligableExchangeInput;
+  // TODO: Add exchange rates?
+  mapping (address => bool) public isEligableExchangeOutput;
+
+  constructor(address _dsProxyCacheAddr) DSProxy(_dsProxyCacheAddr) public {
+  }
+
+  function updateExchangeInput( address _token
+                              , bool _isTrusted ) public
+			                            auth
+                                                      returns (bool success) {
+
+    isEligableExchangeInput[_token] = _isTrusted;
+    return true;
+  }
+
+  function updateExchangeOutput( address _token
+                               , bool _isTrusted ) public
+			                             auth
+                                                       returns (bool success) {
+
+    isEligableExchangeOutput[_token] = _isTrusted;
+    return true;
+  }
+
+  function isExchangeApproved( address _offeredToken
+                             , address _desiredToken ) public view
+			                                 returns (bool) {
+
+    return
+      isEligableExchangeInput[_offeredToken]
+      && isEligableExchangeOutput[_desiredToken];
+  }
+
+  modifier _exchangePrerequisites( address _offeredToken
```

Comment:
I would maybe just inline these two calls to `exchangeTransfer` and `exchangeApprove`. Having it as a modifier breaks up the control flow and makes it a little harder to parse IMO.

Reply:
Fair enough!